test/externalTests/foundry.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import os
+import re
+from pathlib import Path
+from shutil import which
+from string import Template
+from typing import Tuple
+import requests
+
+from external_test import AVAILABLE_PRESETS
+from external_test import settings_from_preset, run_cmd
+from external_test import TestConfig, TestRunner, ExternalTest
+
+# TODO: Add hardhat runner
+# TODO: Add truffle runner
+
+class FoundryRunner(TestRunner):
+    """Configure and run Foundry-based projects"""
+
+    profile_tmpl = Template("""
+[profile.${name}]
+gas_reports = [\"*\"]
+auto_detect_solc = false
+solc = \"${solc}\"
+evm_version = \"${evm_version}\"
+optimizer = ${optimizer}
+via_ir = ${via_ir}
+[profile.${name}.optimizer_details]
+yul = ${yul}
+""")
```

Comment:
The formatting is off in quite a few files. Maybe we can install a formatting tool, and run a check during every push?

Reply:
Yeah, we could use `black`. Though I wouldn't necessarily add it here. We should have it enabled for all Python scripts. mypy as well. For this PR I'd just format it properly manually.